opal/mca/threads/pthreads/threads_pthreads_yield.c

```diff
@@ -0,0 +1,54 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil -*- */
+/*
+ * Copyright (c) 2020      High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ *
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include <unistd.h>
+#include <time.h>
+#ifdef HAVE_SCHED_H
+#include <sched.h>
+#endif
+
+#include "opal/mca/threads/threads.h"
+#include "opal/mca/threads/pthreads/threads_pthreads.h"
+#include "opal/sys/atomic.h"
+
+
+opal_threads_pthreads_yield_strategy_t opal_threads_pthreads_yield_strategy = OPAL_PTHREADS_YIELD_SCHED_YIELD;
+uint64_t opal_threads_pthreads_yield_usleep = 1;
+uint64_t opal_threads_pthreads_yield_nsleep = 100;
+
+
+void opal_thread_yield(void)
```

Comment:
It seems like we can do a little better here.  If opal_thread_yield() were a function pointer to the right case, we could get 1 function call and the case statement out of the critical path.

Reply:
Done, the yield strategy in use is now a function pointer and the call to `opal_thread_yield` can be inlined so all that's left is a call through that pointer.